File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/services/HostedCheckoutServiceInterface.java
package com.onlinepayments.example.service.interfaces.services;

import com.onlinepayments.example.service.dtos.hostedcheckout.CreateHostedCheckoutRequestDto;
import com.onlinepayments.example.service.dtos.hostedcheckout.CreateHostedCheckoutResponseDto;
import com.onlinepayments.example.service.dtos.paymentbyhostedcheckoutid.GetPaymentByHostedCheckoutIdResponseDto;

public interface HostedCheckoutServiceInterface {
    CreateHostedCheckoutResponseDto createHostedCheckout(CreateHostedCheckoutRequestDto request);

    GetPaymentByHostedCheckoutIdResponseDto getPaymentByHostedCheckoutId(String id);
}

File: server/nodejs/src/business/interfaces/hosted-checkout-service-interface.ts
import { FastifyBaseLogger } from 'fastify';
import { CreateHostedCheckoutRequestDto } from '../dtos/hosted-checkout/create-hosted-checkout-request-dto';
import { CreateHostedCheckoutResponseDto } from '../dtos/hosted-checkout/create-hosted-checkout-response-dto';
import { IHostedCheckoutClient } from '../../infrastructure/interfaces/hosted-checkout-client-interface';
import { GetPaymentByHostedCheckoutResponseDto } from '../dtos/hosted-checkout/get-payment-by-hosted-checkout-response-dto';

export interface IHostedCheckoutService {
    createHostedCheckout(
        client: IHostedCheckoutClient,
        requestDto: CreateHostedCheckoutRequestDto,
        logger: FastifyBaseLogger,
    ): Promise<CreateHostedCheckoutResponseDto>;

    getPaymentByHostedCheckoutId(
        client: IHostedCheckoutClient,
        hostedCheckoutId: string,
        logger: FastifyBaseLogger,
    ): Promise<GetPaymentByHostedCheckoutResponseDto | null>;
}
